src/librustdoc/clean/types.rs

```diff
@@ -1538,6 +1538,41 @@ impl From<ast::FloatTy> for PrimitiveType {
     }
 }
 
+impl From<ty::IntTy> for PrimitiveType {
+    fn from(int_ty: ty::IntTy) -> PrimitiveType {
+        match int_ty {
+            ty::IntTy::Isize => PrimitiveType::Isize,
+            ty::IntTy::I8 => PrimitiveType::I8,
+            ty::IntTy::I16 => PrimitiveType::I16,
+            ty::IntTy::I32 => PrimitiveType::I32,
+            ty::IntTy::I64 => PrimitiveType::I64,
+            ty::IntTy::I128 => PrimitiveType::I128,
+        }
+    }
+}
+
+impl From<ty::UintTy> for PrimitiveType {
+    fn from(uint_ty: ty::UintTy) -> PrimitiveType {
+        match uint_ty {
+            ty::UintTy::Usize => PrimitiveType::Usize,
+            ty::UintTy::U8 => PrimitiveType::U8,
+            ty::UintTy::U16 => PrimitiveType::U16,
+            ty::UintTy::U32 => PrimitiveType::U32,
+            ty::UintTy::U64 => PrimitiveType::U64,
+            ty::UintTy::U128 => PrimitiveType::U128,
+        }
+    }
+}
+
+impl From<ty::FloatTy> for PrimitiveType {
+    fn from(float_ty: ty::FloatTy) -> PrimitiveType {
+        match float_ty {
+            ty::FloatTy::F32 => PrimitiveType::F32,
+            ty::FloatTy::F64 => PrimitiveType::F64,
+        }
+    }
+}
+
```

Comment:
Where are these used?

Reply:
I know at least one of them is used, because I had an error. I decided to implement From for all three types because it was implemented for the `ast` ones.

Reply:
I would prefer to implement it for as few as possible actually, since the compiler has a limitation that it doesn't warn on dead code for impls even if one of the types is private. Do you mind only adding the one that gave the error (and pasting the error if possible - it's still not clear to me why this is needed since no other part of rustdoc changed)?

Reply:
> it's still not clear to me why this is needed since no other part of rustdoc changed

Consider something like:
```rust
match ty.kind() {
    ty::Int(ity) => whatever(ity.into()),
    _ => {}
}
```

With this PR, there are no changes to the code, but `ity` changed from an `ast::IntTy` to a `ty::IntTy`, and we need to have a `From` implementation for it.

Reply:
Got it, thanks. Can you remove the impls for `ast::IntTy` then? IIUC they should no longer be needed.

Reply:
FYI, here is the full output when removing the impls:

<details><summary>Output</summary>

```
error[E0277]: the trait bound `clean::types::PrimitiveType: From<rustc_middle::ty::IntTy>` is not satisfied
    --> src/librustdoc/clean/mod.rs:1541:42
     |
1541 |             ty::Int(int_ty) => Primitive(int_ty.into()),
     |                                          ^^^^^^^^^^^^^ the trait `From<rustc_middle::ty::IntTy>` is not implemented for `clean::types::PrimitiveType`
     |
     = help: the following implementations were found:
               <clean::types::PrimitiveType as From<rustc_hir::PrimTy>>
     = note: required because of the requirements on the impl of `Into<clean::types::PrimitiveType>` for `rustc_middle::ty::IntTy`

error[E0277]: the trait bound `clean::types::PrimitiveType: From<rustc_middle::ty::UintTy>` is not satisfied
    --> src/librustdoc/clean/mod.rs:1542:44
     |
1542 |             ty::Uint(uint_ty) => Primitive(uint_ty.into()),
     |                                            ^^^^^^^^^^^^^^ the trait `From<rustc_middle::ty::UintTy>` is not implemented for `clean::types::PrimitiveType`
     |
     = help: the following implementations were found:
               <clean::types::PrimitiveType as From<rustc_hir::PrimTy>>
     = note: required because of the requirements on the impl of `Into<clean::types::PrimitiveType>` for `rustc_middle::ty::UintTy`

error[E0277]: the trait bound `clean::types::PrimitiveType: From<rustc_middle::ty::FloatTy>` is not satisfied
    --> src/librustdoc/clean/mod.rs:1543:46
     |
1543 |             ty::Float(float_ty) => Primitive(float_ty.into()),
     |                                              ^^^^^^^^^^^^^^^ the trait `From<rustc_middle::ty::FloatTy>` is not implemented for `clean::types::PrimitiveType`
     |
     = help: the following implementations were found:
               <clean::types::PrimitiveType as From<rustc_hir::PrimTy>>
     = note: required because of the requirements on the impl of `Into<clean::types::PrimitiveType>` for `rustc_middle::ty::FloatTy`

error[E0277]: the trait bound `clean::types::PrimitiveType: From<rustc_ast::IntTy>` is not satisfied
    --> src/librustdoc/clean/types.rs:1510:48
     |
1510 |             hir::PrimTy::Int(int_ty) => int_ty.into(),
     |                                                ^^^^ the trait `From<rustc_ast::IntTy>` is not implemented for `clean::types::PrimitiveType`
     |
     = help: the following implementations were found:
               <clean::types::PrimitiveType as From<rustc_hir::PrimTy>>
     = note: required because of the requirements on the impl of `Into<clean::types::PrimitiveType>` for `rustc_ast::IntTy`

error[E0277]: the trait bound `clean::types::PrimitiveType: From<rustc_ast::UintTy>` is not satisfied
    --> src/librustdoc/clean/types.rs:1511:51
     |
1511 |             hir::PrimTy::Uint(uint_ty) => uint_ty.into(),
     |                                                   ^^^^ the trait `From<rustc_ast::UintTy>` is not implemented for `clean::types::PrimitiveType`
     |
     = help: the following implementations were found:
               <clean::types::PrimitiveType as From<rustc_hir::PrimTy>>
     = note: required because of the requirements on the impl of `Into<clean::types::PrimitiveType>` for `rustc_ast::UintTy`

error[E0277]: the trait bound `clean::types::PrimitiveType: From<rustc_ast::FloatTy>` is not satisfied
    --> src/librustdoc/clean/types.rs:1512:54
     |
1512 |             hir::PrimTy::Float(float_ty) => float_ty.into(),
     |                                                      ^^^^ the trait `From<rustc_ast::FloatTy>` is not implemented for `clean::types::PrimitiveType`
     |
     = help: the following implementations were found:
               <clean::types::PrimitiveType as From<rustc_hir::PrimTy>>
     = note: required because of the requirements on the impl of `Into<clean::types::PrimitiveType>` for `rustc_ast::FloatTy`

error: aborting due to 6 previous errors
```

</details>

I thought we could remove some, but in fact, seems like we still need all 6 `From` implementations 🤷‍♂️

Reply:
Hmm, ok. I have some hope this can be cleaned up but it doesn't need to happem in this PR, I'll take a look after it's merged.